restli-server/src/main/java/com/linkedin/restli/internal/server/util/RestUtils.java

```diff
@@ -251,6 +250,36 @@ public static String pickBestEncoding(String acceptHeader, Set<String> customMim
    */
   public static DataMap projectFields(final DataMap dataMap, final ProjectionMode projectionMode,
       final MaskTree projectionMask)
+  {
```

Comment:
Do you think since this adding field is in response, maybe this `projectFields` should be in Response Util class instead of Rest Utils?

Reply:
We can move, unfortunately this is a public API and so we cannot delete it. 

If this is not a big concern, I prefer to do this refactoring in a different PR. The current PR is already large.